client/python/examples/async_logging.py

```diff
@@ -0,0 +1,210 @@
+"""
+Demonstrates how to run jobs, and also log all
+changes to that job or job_set in realtime concurrently.
+"""
+
+import os
+import threading
+import time
+import uuid
+
+import grpc
+from armada_client.client import ArmadaClient
+from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
+from armada_client.k8s.io.apimachinery.pkg.api.resource import (
+    generated_pb2 as api_resource,
+)
+
+
+def create_dummy_job(client):
+    """
+    Create a dummy job with a single container.
+    """
+
+    # For infomation on where this comes from,
+    # see: https://github.com/kubernetes/api/blob/master/core/v1/generated.proto
+    pod = core_v1.PodSpec(
+        containers=[
+            core_v1.Container(
+                name="container1",
+                image="index.docker.io/library/ubuntu:latest",
+                args=["sleep", "10s"],
+                securityContext=core_v1.SecurityContext(runAsUser=1000),
+                resources=core_v1.ResourceRequirements(
+                    requests={
+                        "cpu": api_resource.Quantity(string="120m"),
+                        "memory": api_resource.Quantity(string="510Mi"),
+                    },
+                    limits={
+                        "cpu": api_resource.Quantity(string="120m"),
+                        "memory": api_resource.Quantity(string="510Mi"),
+                    },
+                ),
+            )
+        ],
+    )
+
+    return [client.create_job_request_item(priority=1, pod_spec=pod)]
+
+
+def watch_job_set(client: ArmadaClient, queue: str, job_set_id):
```

Comment:
I don't love the infinite function here but I don't know if you need to change.

Reply:
Would putting a timer on it be effective? Like maybe 30 seconds - a minute?

Reply:
We will see what others think.  A timer would be good but I may not be necessary for this

Reply:
The infinte function will be removed and replaced with something else.